### PR TITLE
OCPBUGS-17989: pkg/psalabelsyncer: enforce syncing in case label is set

### PR DESCRIPTION
**Pull request by s-urbaniak**
As discussed with @stlaz this fixes syncing PSa labels if `security.openshift.io/scc.podSecurityLabelSync` is set to `true` by the user.

Currently this case is ommitted if managed fields are set to cluster policy controller.

/cc @deads2k @ibihim @liouk @stlaz

### CODE REVIEW

File: pkg/psalabelsyncer/podsecurity_label_sync_controller.go [hunk 1]
```diff
@@ -348,16 +348,17 @@ func (c *PodSecurityAdmissionLabelSynchronizationController) syncNamespace(ctx c
 	}
 
 	var shouldUpdate bool
+	mustSync := ns.Labels[labelSyncControlLabel] == "true"
```
Comment (ibihim) [inline pkg/psalabelsyncer/podsecurity_label_sync_controller.go:351]: We can stop immediately, if we shall not sync:

```suggestion
	mustSync := ns.Labels[labelSyncControlLabel] == "true"

	if mustNotSync := ns.Labels[labelSyncControlLabel] == "false"; mustNotSync {
	    return nil
	}
```

File: pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go [hunk 2]
```diff
@@ -362,18 +405,21 @@ func TestEnforcingPodSecurityAdmissionLabelSynchronizationController_sync(t *tes
 			"controlled-namespace-previous-enforce-labels/testspecificsa3":                  sets.NewString("scc_restricted"),
 			"controlled-namespace-previous-warn-labels/testspecificsa3":                     sets.NewString("scc_restricted"),
 			"controlled-namespace-previous-enforce-version-different-owner/testspecificsa3": sets.NewString("scc_restricted"),
+			"controlled-namespace-different-owner-and-sync-label-true/testspecificsa3":      sets.NewString("scc_restricted"),
+			"controlled-namespace-different-owner-and-sync-label-unset/testspecificsa3":     sets.NewString("scc_restricted"),
```
Comment (ibihim) [inline pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go:409]: I have not enough insights into the test cases, but don't we need to list: `non-controlled-namespace-different-owner-and-sync-label-false`?

File: pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go [hunk 3]
```diff
@@ -40,6 +40,49 @@ func testNamespaces() []*corev1.Namespace {
...
+				psapi.EnforceVersionLabel:                        "bogus enforce version value",
+				"security.openshift.io/scc.podSecurityLabelSync": "true",
+			},
+			ManagedFields: managedLabelsFields("someone-else",
+				psapi.EnforceLevelLabel,
+				psapi.EnforceVersionLabel,
+			),
+		}},
+		{ObjectMeta: metav1.ObjectMeta{
+			Name:        "non-controlled-namespace-different-owner-and-sync-label-false",
```
Comment (ibihim) [inline pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go:57]: What is the meaning of `non-controlled` and `controlled`? The only difference between this and the namespace above is the label sync value, which is mentioned explicitly at the end of the namespace's name.

File: pkg/psalabelsyncer/podsecurity_label_sync_controller.go (hunk 1 above)
Comment (s-urbaniak) [reply pkg/psalabelsyncer/podsecurity_label_sync_controller.go:351]: I would advice against adding a second flow next to `shouldUpdate`. This is adding unnecessary complexity.

File: pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go (hunk 3 above)
Comment (s-urbaniak) [reply pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go:57]: as far as i see this denotes what namespaces are "controlled" or "not controlled" by the policy controller.

File: pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go (hunk 2 above)
Comment (s-urbaniak) [reply pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go:409]: No we don't need to. As this namespace is non-controlled, we don't need a service account fixture.

File: pkg/psalabelsyncer/podsecurity_label_sync_controller.go (hunk 1 above)
Comment (s-urbaniak) [reply pkg/psalabelsyncer/podsecurity_label_sync_controller.go:351]: additionally your suggestion does not pass unit tests.

File: pkg/psalabelsyncer/podsecurity_label_sync_controller.go [hunk 4]
```diff
@@ -371,7 +372,7 @@ func (c *PodSecurityAdmissionLabelSynchronizationController) syncNamespace(ctx c
 		return nil
 	}
 
-	_, err = c.namespaceClient.Apply(ctx, nsApplyConfig, metav1.ApplyOptions{FieldManager: controllerName})
+	_, err = c.namespaceClient.Apply(ctx, nsApplyConfig, metav1.ApplyOptions{FieldManager: controllerName, Force: true})
```
[the remark this answers is not in the document]
Comment (s-urbaniak) [reply pkg/psalabelsyncer/podsecurity_label_sync_controller.go:375]: `Force: mustSync` sgtm 👍

File: pkg/psalabelsyncer/podsecurity_label_sync_controller.go (hunk 1 above)
Comment (ibihim) [reply pkg/psalabelsyncer/podsecurity_label_sync_controller.go:351]: This is strange, but I guess it is due to the state of the unit tests?

File: pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go (hunk 2 above)
Comment (ibihim) [reply pkg/psalabelsyncer/podsecurity_label_sync_controller_test.go:409]: Thx for the explanation.

**Pull request merged**